src/analysis/ASTnodefunctions.ts

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import {
+    DocumentNode,
+    FieldNode,
+    SelectionSetNode,
+    DefinitionNode,
+    Kind,
+    SelectionNode,
+    ArgumentNode,
+} from 'graphql';
+
+// TODO: handle variables and arguments
+// ! this is not functional
+const getArgObj = (args: ArgumentNode[]): { [index: string]: any } => {
```

Comment:
Related to previous comments but this function may become obsolete and the weight function will handle parsing `ArgumentNode`s

Reply:
Yes, none of this is functional. I left the code in case it became useful.